### PR TITLE
Add additional WooCommerce onboarding styling to connection flow

### LINKED ISSUE #2152: Onboarding - Profile Wizard: Step 3 - Jetpack Connection Flow
This part of the profiler lives on WordPress.com, and will need to be implemented in `wp-calypso`.

The relevant code for the Jetpack connection process lives at https://github.com/Automattic/wp-calypso/blob/master/client/blocks/login/index.jsx. We will need to work with Calypso/Jetpack teams to inject our own styling and copy. It looks like there is prior art here already to work off of.

The user will also need to be properly redirected after logging in and connecting. They should end up on Step 4 - Store Address.

[Master Thread](https://wp.me/p90Yrv-10i) | [Designs](https://wp.me/p6riRB-4rG)

### LINKED ISSUE #2166: Onboarding - Profile Wizard: Implement Muriel styling for form components
The profile wizard uses Muriel components for form inputs, buttons, etc. These components exist in Gutenberg, but we need to implement our own layer of styling over these.

See p1556119186006300-slack-wc-onboarding for Figma links containing more information on the styles.

We will need styles for:

- [x] Primary Button
- [x] Secondary Button
- [x] Checkboxes
- [ ] Form Inputs
- [x] Dropdowns

In terms of implementation, we should be overriding these styles at the component level. See p1557235811036900-slack-nux. Newspack is also doing something similar. Example: https://github.com/Automattic/newspack-plugin/pull/46.

On hold pending: https://github.com/woocommerce/woocommerce-admin/issues/2166#issuecomment-493434379

### PR DESCRIPTION
**Pull request by emdashcodes**
This PR implements the remaining missing styles for the WooCommerce/Jetpack onboarding connection flow. This includes the white header bar, width/spacing changes, link and color treatments, and the largest visual change: Muriel/Material styled form components.

It is only enabled in development, is only active through a specific connection flow, and builds upon the previously merged PRs: https://github.com/Automattic/wp-calypso/pull/32993 and https://github.com/Automattic/wp-calypso/pull/33073. 

This PR implements a temporary Muriel styled `TextControl` component in `client/extensions/woocommerce/components` that matches the interface of `@wordpress/components`'s `TextControl` and the `calypso-ui` component proposed at https://github.com/Automattic/wp-calypso/pull/34277. This PR can be switched over to use the `calypso-ui` component, once it is unblocked by https://github.com/WordPress/gutenberg/pull/16227. (The login code is loaded via SSR). This temporary component allows us to test the styles all together, and allows me to keep iterating unblocked on the WooCommerce.com styles next (which are also hosted in Calypso).

Closes https://github.com/woocommerce/woocommerce-admin/issues/2166. Closes https://github.com/woocommerce/woocommerce-admin/issues/2152.

#### Screenshots

Screen Shot 2019-06-28 at 12 03 49 PM

Screen Shot 2019-06-28 at 12 01 42 PM

#### Testing instructions

* Disconnect Jetpack from your site
* `npm install` to get the updated `color-studio` package
* `npm start`

--

If you have WooCommerce / WooCommerce Admin Installed:

* Enable the onboarding flow in `WooCommerce Admin` and to tell it to point at local Calypso. 
    * Start WooCommerce in development (not plugin) mode
    * Define the following constant in your `wp-config.php` or a plugin file: `define( 'WOOCOMMERCE_CALYPSO_LOCAL', true );`.
    * Make a `POST` request to `/wp-json/wc-admin/v1/onboarding/profile` on your local site, setting `skipped` and `completed` to false.
* Visit `/wp-admin/admin.php?page=wc-admin` on your local site, and the onboarding wizard should be displayed.
* Click `Get Started`, and then `Activate & Continue`.

If you don't want to install WooCommerce Admin:

* Disconnect Jetpack from your site
* Click “Setup Jetpack”.
* On the connection/authorization screen ( https://wordpress.com/jetpack/connect/authorize?client_id= ….. ) copy and paste the whole URL.
* Replace `https://wordpress.com` with ` http://calypso.localhost:3000` and replace the `from` parameter in the URL (it may be something like `connection-banner` or `landing-page-bottom`) with  `from=woocommerce-setup-wizard` . See Pastebin `213d5`  or ping me if you have issues.

---

* Walk through the login and sign-up flows. View both views via the link in the footer. Test typing in the inputs.

* Open `config/development.json` and disable the `jetpack/connect/woocommerce` flag. Refresh the login and sign-up pages and verify the existing Jetpack connection flow appears.

### CODE REVIEW

File: client/blocks/login/index.jsx
Comment (a8ci18n) [inline client/blocks/login/index.jsx]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Log in to your WordPress.com account to set up Jetpack.' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/jetpack-connect/auth-form-header.jsx
Comment (a8ci18n) [inline client/jetpack-connect/auth-form-header.jsx]: :disappointed: 19 existing translations will be lost with this change.

Comment (justinshreve) [reply client/jetpack-connect/auth-form-header.jsx]: I was originally changing these strings to match the mocks, but I think they are similar enough that we should stick with the existing.

File: client/blocks/login/login-form.jsx
Comment (simison) [inline client/blocks/login/login-form.jsx]: Should you use `ExternalLink` component here?

File: client/jetpack-connect/style.scss
Comment (simison) [inline client/jetpack-connect/style.scss]: Please use CSS colour variables instead of hex codes.

See other files for examples or https://github.com/Automattic/wp-calypso/blob/3e69a83a497e8e2c0708d074311398cd98037df0/packages/calypso-color-schemes/src/shared/_color-schemes.scss for the full list.

E.g. this would be `var( --color-white )`

File: client/extensions/woocommerce/components/text-control/style.scss
Comment (simison) [inline client/extensions/woocommerce/components/text-control/style.scss]: If this text input is different from Calypso's and could be re-usable anywhere where Muriel components are needed, could there be a centralized repository where this can be pulled from? Maybe in `packages/` in Calypso?

Is the same component used in WooCommerce and thus now duplicate code?

Comment (simison) [reply client/extensions/woocommerce/components/text-control/style.scss]: Poking around, there might not be any other place to include this for now. :-)

File: client/blocks/login/index.jsx
Comment (simison) [inline client/blocks/login/index.jsx]: Humm, I wonder if it's correct to say "Jetpack account" instead of "WordPress.com account"? I get it's prolly to lessen brand confusion but wordpress.com will be there no matter what (in the URL if not elsewhere). What if they already have one but don't realize that now with this copy changed?

File: package.json
Comment (simison) [inline package.json]: Looks like we could pull 1.0.5 directly from NPM :tada:

https://github.com/Automattic/color-studio/blob/master/CHANGELOG.md#105

Which changes were you interested in specifically?

**Comment by simison** [comment]
I added e2e labels to just do a test run now; feel free to remove them once they're done. Good to re-run them before merging once more, tho. :-)

**Review by simison — COMMENTED** [review]
I've tested that these changes work fine with Jetpack connect flow 👍

**Comment by simison** [comment]
BTW did I say yet that I love the new styles here and wish we could Murielize all signup flows soon! 💪 

Jetpack test failures seemed unrelated (fix in https://github.com/Automattic/wp-calypso/pull/34401)

Didn't look into failing wp-desktop tests.

File: package.json
Comment (justinshreve) [reply package.json]: The earlier versions did not include the WooCommerce Purple / other brand colors. Thanks for pointing out this is on npm now! Updated.

File: client/blocks/login/index.jsx
Comment (justinshreve) [reply client/blocks/login/index.jsx]: Yup, I agree. I originally changed these in the WC flow to match the mocks, but I think it would be better to use the same language (and it's already translated). Fixed.

File: client/extensions/woocommerce/components/text-control/style.scss
Comment (justinshreve) [reply client/extensions/woocommerce/components/text-control/style.scss]: Yes, we do want to pull this from a shared repository. The goal is to put this in `calypso-ui`, based off of the `@wordpress/component` version. I have a PR open for this https://github.com/Automattic/wp-calypso/pull/34277. However, we are currently blocked until we help push through a Gutenberg PR linked in the description above.

I've made sure to use the same interface so that we can delete this temporary component later and switch to the other one.

I've added some comments to the code to be clearer about this.

File: client/jetpack-connect/style.scss
Comment (justinshreve) [reply client/jetpack-connect/style.scss]: Thanks for catching these two that I missed. Updated.

File: client/blocks/login/login-form.jsx
Comment (justinshreve) [reply client/blocks/login/login-form.jsx]: Done.

File: client/extensions/woocommerce/components/text-control/style.scss
Comment (simison) [reply client/extensions/woocommerce/components/text-control/style.scss]: Fabulous! :-) @ockham pointed out me the plan and it sure sounds great!

**Comment by justinshreve** [comment]
Thanks for the PR feedback and for looking at the Jetpack test failures!

It looks like other tests are failing now pulling in the new color-studio version. `Invalid Version: github:automattic/color-studio#2569acb85214ac2413ab5e8ac7b87b04227ba4c4`

I wonder if maybe something is cached on circleci? It shouldn't be pulling from GitHub now. I'll try rerunning them a bit later.

**Comment by simison** [comment]
>It looks like other tests are failing now pulling in the new color-studio version. Invalid Version: github:automattic/color-studio#2569acb85214ac2413ab5e8ac7b87b04227ba4c4

Likely unsynced `npm-shrinkwrap.json`? You can test locally by running `npm ci`

File: client/blocks/login/login-form.jsx
Comment (timmyc) [inline client/blocks/login/login-form.jsx]: Does this need to be included here and in line 279 also?

File: client/extensions/woocommerce/components/text-control/index.js
Comment (timmyc) [inline client/extensions/woocommerce/components/text-control/index.js]: I like where you put this component, good ol days.

**Review by timmyc — APPROVED** [review]
This is testing out well for me.

File: client/blocks/login/login-form.jsx
Comment (justinshreve) [reply client/blocks/login/login-form.jsx]: Thanks for catching that, fixed!

**Pull request merged**
